Explorer/Assets/Scripts/ECS/Unity/Materials/Systems/CreateMaterialSystemBase.cs

```diff
@@ -0,0 +1,81 @@
+using Arch.Core;
+using ECS.Abstract;
+using ECS.StreamableLoading.Components;
+using ECS.StreamableLoading.Components.Common;
+using ECS.Unity.Textures.Components;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace ECS.Unity.Materials.Systems
+{
+    public abstract class CreateMaterialSystemBase : BaseUnityLoopSystem
+    {
+        private readonly int attemptsCount;
+        protected readonly IMaterialsCache materialsCache;
+
+        internal Material sharedMaterial { get; private set; }
+
+        protected CreateMaterialSystemBase(World world, IMaterialsCache materialsCache, int attemptsCount) : base(world)
+        {
+            this.materialsCache = materialsCache;
+            this.attemptsCount = attemptsCount;
+        }
+
+        internal abstract string materialPath { get; }
+
+        public override void Initialize()
+        {
+            sharedMaterial = Resources.Load<Material>(materialPath);
+            Assert.IsNotNull(sharedMaterial);
+        }
+
+        protected Material CreateNewMaterialInstance() =>
+            new (sharedMaterial);
+
+        protected bool TryCreateGetTextureIntention(in TextureComponent? textureComponent, ref EntityReference entityReference)
+        {
+            if (textureComponent == null) return false;
+
+            Entity entity = World.Create(new GetTextureIntention
+            {
+                CommonArguments = new CommonLoadingArguments(textureComponent.Value.Src, attempts: attemptsCount),
+                WrapMode = textureComponent.Value.WrapMode,
+                FilterMode = textureComponent.Value.FilterMode,
+            });
+
+            entityReference = World.Reference(entity);
+            return true;
+        }
+
+        protected void DestroyEntityReference(in EntityReference entityReference)
+        {
+            if (!entityReference.IsAlive(World))
+                return;
+
+            World.Destroy(entityReference);
```

Comment:
WDYT about adding a `DeleteEntityIntention`, and have just one system destroy everything that needs to be destroyed?

Reply:
It's a good idea, we still don't have a system to destroy entities?! Do we?